xdsl/dialects/complex.py

```diff
@@ -216,12 +317,40 @@ def __init__(self, value: ArrayAttr, result_type: ComplexType):
     def get_constant_value(self) -> Attribute:
         return self.value
 
+    @staticmethod
+    def from_tuple_and_width(
+        value: tuple[int | float, int | float], width: int
+    ) -> ConstantOp:
+        if width == 1 and isa(value, tuple[int, int]):
```

Comment:
This is a little scary, since the Python type system actually treats int as a subtype of float, which complicates the check for ints here. I would recommend passing the type directly instead of a width, as in rewrites it's already available on an operand, and during creation it's usually available as `f32` or whatever.